test_simple_graph.py

```diff
@@ -3,207 +3,257 @@
 import pytest
 
 
-def test_init():
+@pytest.fixture(scope='function')
+def empty_graph():
+    ''' Define an empty graph. '''
+    return Graph()
+
+
+@pytest.fixture(scope='function')
+def non_empty_graph():
+    ''' Define an graph with a node. '''
+    g = Graph()
+    g.add_node("A")
+    return g
+
+
+@pytest.fixture(scope='function')
+def edges_graph():
+    ''' Define a graph with an edge. '''
+    g = Graph()
+    g.add_edge("A", "B")
+    return g
+
+
+def test_init(empty_graph):
     ''' Test graph creation. '''
-    assert Graph().graph == {}
+    g = empty_graph
+    assert g.graph == {}
 
 
-def test_add_node():
+def test_add_node_empty(empty_graph):
     ''' Test node adding. '''
-    g = Graph()
     # Add to empty graph
+    g = empty_graph
     g.add_node("A")
     assert "A" in g.graph
     assert g.graph["A"] == []
 
+
+def test_add_node_non_empty(non_empty_graph):
+    ''' Test node adding. '''
+    g = non_empty_graph
     # Add to populated graph
     g.add_node("B")
     assert "B" in g.graph
     assert g.graph["B"] == []
 
 
-def test_nodes():
+def test_nodes_empty(empty_graph):
     ''' Test node listing. '''
-    g = Graph()
+    g = empty_graph
     # Test empty graph
     assert g.nodes() == []
-    g.add_node("A")
+
+
+def test_nodes_non_empty(non_empty_graph):
+    ''' Test node listing. '''
+    g = Graph()
     g.add_node("B")
     # Test populated graph
     assert "A" and "B" in g.graph
```

Comment:
This test does not do what you want to do.  It's asserting that "A" is `True` (which it is) and that "B" is in the graph.